cmd/dvotenode/dvotenode.go

```diff
@@ -561,56 +582,48 @@ func main() {
 				go kk.RevealUnpublished()
 			}
 
-			// Start ethereum events (if web3 endpoint configured)
-			if globalCfg.W3Config.W3External != "" {
-				var evh []ethevents.EventHandler
-				var w3uri string
-				switch {
-				case strings.HasPrefix(globalCfg.W3Config.W3External, "ws"):
-					w3uri = globalCfg.W3Config.W3External
-				case strings.HasSuffix(globalCfg.W3Config.W3External, "ipc"):
-					w3uri = globalCfg.W3Config.W3External
-				default:
-					log.Fatal("web3 external must be websocket or IPC for event subscription")
-				}
+			// Start ethereum events (if at least one web3 endpoint configured)
 
-				evh = append(evh, ethevents.HandleVochainOracle)
-
-				var initBlock *int64
-				if !globalCfg.EthEventConfig.SubscribeOnly {
-					initBlock = new(int64)
-					chainSpecs, err := ethchain.SpecsFor(globalCfg.W3Config.ChainType)
-					if err != nil {
-						log.Warn("cannot get chain block to start looking for events, using 0")
-						*initBlock = 0
-					} else {
-						*initBlock = chainSpecs.StartingBlock
-					}
-				}
+			var evh []ethevents.EventHandler
+			evh = append(evh, ethevents.HandleVochainOracle)
 
-				whiteListedAddr := []string{}
-				for _, addr := range globalCfg.VochainConfig.EthereumWhiteListAddrs {
-					if ethcommon.IsHexAddress(addr) {
-						whiteListedAddr = append(whiteListedAddr, addr)
-					}
+			var initBlock int64
+			if !globalCfg.EthEventConfig.SubscribeOnly {
+				chainSpecs, err := ethchain.SpecsFor(globalCfg.W3Config.ChainType)
+				if err != nil {
+					log.Fatal("cannot get chain block to start looking for events")
 				}
-				if len(whiteListedAddr) > 0 {
-					log.Infof("ethereum whitelisted addresses %+v", whiteListedAddr)
+				initBlock = chainSpecs.StartingBlock
+			}
+
+			whiteListedAddr := []string{}
+			for _, addr := range globalCfg.VochainConfig.EthereumWhiteListAddrs {
+				if addr == "[]" {
+					// avoid empty address and pflag autofill in case of empty array
+					continue
 				}
-				if err := service.EthEvents(
-					context.Background(),
-					w3uri,
-					globalCfg.W3Config.ChainType,
-					initBlock,
-					cm,
-					signer,
-					vnode,
-					evh,
-					sc,
-					whiteListedAddr); err != nil {
-					log.Fatal(err)
+				if ethcommon.IsHexAddress(addr) {
+					whiteListedAddr = append(whiteListedAddr, addr)
+				} else {
+					log.Warnf("whitelist address %s is not a valid hex address", addr)
 				}
 			}
+			if len(whiteListedAddr) > 0 {
+				log.Infof("ethereum whitelisted addresses %+v", whiteListedAddr)
+			}
+			if err := service.EthEvents(
+				context.Background(),
+				w3uris,
+				globalCfg.W3Config.ChainType,
+				initBlock,
+				cm,
+				signer,
+				vnode,
+				evh,
+				sc,
+				whiteListedAddr); err != nil {
```

Comment:
I really think we're going way, way overkill with avoiding long lines. You're literally writing multiple lines which are >80% whitespace :) This entire call can very easily fit in 2-3 lines.